hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -2847,16 +2846,21 @@ private DeleteObjectsResult removeKeysS3(
     }
     if (keysToDelete.isEmpty()) {
       // exit fast if there are no keys to delete
-      return null;
+      return;
     }
     for (DeleteObjectsRequest.KeyVersion keyVersion : keysToDelete) {
       blockRootDelete(keyVersion.getKey());
     }
-    DeleteObjectsResult result = null;
     try {
       if (enableMultiObjectsDelete) {
-        result = deleteObjects(
-            getRequestFactory().newBulkDeleteRequest(keysToDelete, quiet));
+        // Multi object deletion of more than 1000 keys is not supported
+        // by s3. So we are paging the keys by page size whose default
+        // value is 250.
+        for (List<DeleteObjectsRequest.KeyVersion> batchOfKeysToDelete :
```

Comment:
only need to partition when size > that the threshold -avoids building new lists